modules/agent_files/templates/docker-daemon.json.erb

```diff
@@ -0,0 +1,4 @@
+{
+<% if ! scope.function_hiera(["docker::registry_mirror"], "").empty? %> "registry-mirrors": ["<%= scope.function_hiera(["docker::registry_mirror"])%>"],<% end -%>
```

Comment:
I'm unsure about how puppet is going to return the `scope.function_hiera(["docker::registry_mirror"])` being `docker::registry_mirror` an array. Probably worth to take a look once running this to be sure the format (plain string?) is the expected.